serlib/ser_environ.ml

```diff
@@ -73,17 +73,36 @@ module Globals = struct
   include SerType.Pierce(PierceSpec)
 end
 
-type env =
-  [%import: Environ.env]
-  [@@deriving sexp_of]
+module InternalEnv =
+struct
+  type _t = {
+    env_globals       : Globals.t;
+    env_named_context : Constr.named_context;
+    env_rel_context   : Constr.rel_context;
+    env_universes : UGraph.t;
+    env_qualities : Sorts.QVar.Set.t;
+    env_typing_flags  : Declarations.typing_flags;
```

Comment:
Adding to my point in the comment, it seems this structure is already missing two fields that are relevant, in particular `rewrite_rules_allowed` and `symb_pats`, likely `retroknoledge` too?

Reply:
Retroknowledge is garbage for users, this is kernel-facing and (unfortunately) mostly unsound.

As for rewrite rules, the implementation is going to change anyway, and good luck figuring out what to do with the data there. I'm not sure what the point of exposing all of this is. Blindly serializing internal data without even checking that the underlying types have changed is looking for trouble. That's not an API, that's barely better than a memory dump and let the end-user figure out what the pointer addresses are.